src/plugins/discover/public/application/main/state_management/discover_data_state_container.ts

```diff
@@ -249,6 +249,12 @@ export function getDataStateContainer({
             return;
           }
 
+          await profilesManager.resolveDataSourceProfile({
+            dataSource: getAppState().dataSource,
+            dataView: getSavedSearch().searchSource.getField('index'),
```

Comment:
`question`: I couldn't check if it works correctly, does the `dataView` get set correctly when using an ad-hoc data view? Feels that, since the data view is retrieved by `getSavedSearch`, it might not get the current data view.

Reply:
It seems to correctly get the dataView passed to the resolve function 👌

Reply:
Yeah unfortunately it's just the way Discover's state management is currently 😅 There's certainly room for improvement here though...